### ISSUE TITLE
OS X: Shortcut to applications folder in installer

### ISSUE THREAD

**Issue by nullpointoh** [sieren/QSyncthingTray#47]
Just a small UX improvement. Add a shortcut to the apps folder so it's idiot proof to install.

**Comment by sieren** [comment]
Yeah, that'd be nice. I hope someone will create a nice .sh script to do all the work.
Right now I'm just using QT's macdeploy application to automatically create the .dmg.

A nicer and customized dmg file would be lovely, but since I'm still more or less working on QSyncthingTray all by myself, my priorities are elsewhere right now.

**Comment by sieren** [comment]
Maybe worth using https://github.com/LinusU/node-appdmg

**Comment by nullpointoh** [comment]
No worries. If I had any coding skill, I'd help!

On 23 December 2015 at 15:21, Matt Sieren <notifications@github.com> wrote:

> Maybe worth using https://github.com/LinusU/node-appdmg
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sieren/QSyncthingTray/issues/47#issuecomment-166922953>
> .
>



-- 
talk: twitter.com/happydan
read: www.presstojump.com | www.djworx.com
listen: mixcloud.com/happydan

**Comment by xor-gate** [comment]
@sieren I'm maintaining a native tray (written in golang) with fancy dmg for OSX only. Maybe I can try to integrate some of my work into yours because the golang tray is very very minimal and at first sight i'm pretty impressed by your work done with Qt. The only downside is we need a ton of crap to just show a tray icon. Your dmg is 65 mbyte while mine is just 10 mbyte. It seems yours only includes `syncthing-inotify` and the user has to supply `syncthing` binary themself which is a little pain in the ass. Currently how I clay the binaries into one app + dmg is not very nice (shell scripts), but it works.

Have a look:  https://github.com/xor-gate/syncthing-mac/releases

The downside with my OSX `.app` wrapper is I need to spawn actually 3 applications: `syncthing`, `syncthing-tray` and `syncthing-inotify`. See: https://github.com/xor-gate/syncthing-mac/blob/master/syncthing-wrapper.sh.

**Comment by mfk-ableton** [comment]
@xor-gate Thanks :) Down the road I'd want to implement an auto-installer directly in Qt which gets the newest version of Syncthing if none is available.

The bundled `syncthing-inotify` actually has to go, since this is all platform dependent and I dont feel like maintaining bundler-scripts for 3 different platforms right now. (Again this might become an automated auto-download at some point).

As for the dmg-size, thats on my radar too. The 65mb mostly come from the Qt framework which gets bundled inside the .app (or the DLLs on windows respectively). Those can be slimmed down though I suppose by excluding some files. I'll have to investigate that though.

**Comment by sieren** [comment]
@xor-gate Thanks :) Down the road I'd want to implement an auto-installer directly in Qt which gets the newest version of Syncthing if none is available.

The bundled syncthing-inotify actually has to go, since this is all platform dependent and I dont feel like maintaining bundler-scripts for 3 different platforms right now. (Again this might become an automated auto-download at some point).

As for the dmg-size, thats on my radar too. The 65mb mostly come from the Qt framework which gets bundled inside the .app (or the DLLs on windows respectively). Those can be slimmed down though I suppose by excluding some files. I'll have to investigate that though.
